### PR TITLE
librbd: fix snap_exists API overflow issue

### PR DESCRIPTION
**Pull request by xiexingguo**
http://tracker.ceph.com/issues/14129

### CODE REVIEW

**Comment by xiexingguo** [comment]
@dillaman  And this one too. Thank you for you attention.

File: src/include/rbd/librbd.hpp [hunk 1]
```diff
@@ -206,7 +206,9 @@ class CEPH_RBD_API Image
 
   /* snapshots */
   int snap_list(std::vector<snap_info_t>& snaps);
+  /* DEPRECATED; use snap_exists2 */
```
Comment (dillaman) [inline src/include/rbd/librbd.hpp]: Would rather see ```__attribute__ ((deprecated))``` appended to ```snap_exists```.

File: src/librbd/internal.h [hunk 2]
```diff
@@ -127,6 +127,7 @@ namespace librbd {
   void snap_create_helper(ImageCtx *ictx, Context* ctx, const char *snap_name);
   int snap_list(ImageCtx *ictx, std::vector<snap_info_t>& snaps);
   bool snap_exists(ImageCtx *ictx, const char *snap_name);
+  int snap_exists2(ImageCtx *ictx, const char *snap_name, bool *exists);
```
Comment (dillaman) [inline src/librbd/internal.h]: In the internal implementation, just update the ```snap_exists``` signature to return an ```int``` and have ```librbd.cc``` handle the two different APIs by calling the same method.

File: src/test/librbd/test_librbd.cc [hunk 3]
```diff
@@ -668,21 +668,35 @@ TEST_F(TestLibRBD, TestCreateLsDeleteSnapPP)
...
+   
+    int r;
+    bool exists;
+    r = image.snap_exists2("snap1", &exists);
+    ASSERT_EQ(0, r);
+    ASSERT_FALSE(exists);
     ASSERT_EQ(0, image.snap_create("snap1"));
-    ASSERT_TRUE(image.snap_exists("snap1"));
+    r = image.snap_exists2("snap1", &exists);
+    ASSERT_EQ(0, r);
```
Comment (dillaman) [inline src/test/librbd/test_librbd.cc]: Merge this into the previous line: ```ASSERT_EQ(0, image.snap_exists2("snap1", &exists))```

**Comment by xiexingguo** [comment]
@dillaman thank you for your careful review,I will get all these fixed tomorrow. Have a nice day.

**Comment by xiexingguo** [comment]
@dillaman Fixed as you suggested, please review. Thanks a lot!

**Comment by dillaman** [comment]
Couple more spots that should be swapped over to ```snap_exists2``` to avoid the compile warning:

```
./include/rbd/librbd.hpp:210:8: note: declared here
   bool snap_exists(const char *snapname) __attribute__ ((deprecated));
```

**Comment by xiexingguo** [comment]
@dillaman OK~~~, I'll try to find them out as soon as possible:)

**Comment by xiexingguo** [comment]
@dillaman Fixed, please review:)  Thank you for your patience.

**Pull request merged**